### DISCUSSION TITLE
Button with SWT.ARROW style is not styled correctly in dark theme

### DISCUSSION THREAD

**Discussion by dhendriks** [eclipse-platform/eclipse.platform.swt#434] [Q&A]
A `Button` with `SWT.PUSH` style is correctly styled with the dark theme. If I use the `SWT.ARROW` style however, it does not get styled:

![image](https://user-images.githubusercontent.com/375606/195131153-142fd8c5-26cd-4e0a-9821-517a9427901d.png)

I use these in an Eclipse `EditorPart`. I create them as follows: 

```java
Button push = new Button(comp, SWT.PUSH);
Button arrow = new Button(comp, SWT.ARROW | SWT.DOWN);
```

I don't do anything special with them regarding styling. I just set Eclipse to use its built-in dark theme. I use Eclipse 2022-06 on Windows.

Is this an SWT issue? Or is some other part of the Eclipse Platform responsible for this problem?

**Comment by akurtakov** [comment]
This looks like bug in SWT windows port. Ideally one should provide a pure SWT snippet demonstrating the issue to ease further development.

**Reply by dhendriks** [reply]
How can I create a pure SWT snippet that uses the OS dark theme? My Windows is set to a dark theme. I tried using the following, but they don't seem to have an affect:
```
-Dorg.eclipse.swt.display.useSystemTheme=true
-Dorg.eclipse.swt.internal.win32.enableCustomDarkTheme=true
```

**Reply by akurtakov** [reply]
Unfortunately I can't help much with Windows specific things. Eclipse theming engine does a lot of setForeground/Background with hardcoded colors. 
Maybe @niraj-modi or @sravanlakkimsetti can guide you about windows specifics.
